### ISSUE TITLE
Feature Request: Search for grid values

### ISSUE THREAD

**Issue by Maaz4444** [microsoft/vscode-data-wrangler#178]
Type: <b>Bug</b>

# Behaviour

## Expected vs. Actual

XXX

## Steps to reproduce:

1. XXX

# Diagnostic data

-   Jupyter extension version: 2024.4.0
-   Python extension version: 2024.6.0
-   .NET Install Tool for Extension Authors extension version: Not installed
-   Python package dependencies:

```json
{
  "installed": {
    "pandas": "2.2.2"
  },
  "required": {
    "pandas": "0.25.2"
  },
  "unsatisfied": []
}
```

-   Entrypoint: JupyterCellStatusBar
-   Active mode: dataViewer

Extension version: 1.0.2
VS Code version: Code 1.89.0 (b58957e67ee1e712cebf466b995adf4c5307b2bd, 2024-05-01T02:09:22.859Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz (16 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.85GB (0.74GB free)|
|Process Argv|--crash-reporter-id 9e1e643d-73e5-4dab-9dc1-b86bb4fbb721|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
666c4381:30973075
dsvsc021:30996838
da93g388:31013173
pythoncenvptcf:31043158
pythonprc:31039816
dwnewjupytercf:31039676

```

</details>

**Comment by pwang347** [comment]
Hi @Maaz4444, thanks for opening this issue! Just to double check, are you looking to search for a specific value in the grid?

**Comment by Maaz4444** [comment]
Yes, that is correct. I am unable to search any value.

Thanks&Regards,
Maaz Ahmed

**Comment by pwang347** [comment]
Got it, thanks for confirming! I'll keep this open as a feature request to search for values in the grid, since we'll need to make sure it works well for large datasets too.

**Comment by Maaz4444** [comment]
Search, select all, copy
Need all this functionality.


Thanks&Regards,
Maaz Ahmed

**Comment by pwang347** [comment]
Hi @Maaz4444, just wanted to share that search functionality should now be available in pre-release version 1.15.1 - you can access this through the filter menu, which now supports searching by string directly. Hope this helps!

**Comment by Maaz4444** [comment]
Thanks


Cheers,
Maaz Ahmed
